gluten-core/src/main/scala/io/glutenproject/execution/SortExecTransformer.scala

```diff
@@ -52,124 +50,26 @@ case class SortExecTransformer(
 
   override def output: Seq[Attribute] = child.output
 
-  override def outputOrdering: Seq[SortOrder] = sortOrder
+  override def outputOrdering: Seq[SortOrder] = child match {
+    case project: ProjectExecTransformer if project.isPreProject =>
```

Comment:
do we really need `isPreProject` flag ? it seems we can do it for all projects

Reply:
For a normal `ProjectTransformer`, executing this process poses no problems, but the resulting `sortOrder` remains the same as the original.

Reply:
I find a case that before the sort there is a project to rename the sort key, we should not replace alias in this case. So we still need pre-project hint for sort.